crepecake/src/integration-test/java/com/google/cloud/tools/crepecake/registry/DockerCredentialRetrieverIntegrationTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.registry;
+
+import com.google.cloud.tools.crepecake.http.Authorization;
+import java.io.IOException;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+
+/** Integration tests for {@link DockerCredentialRetriever}. */
+public class DockerCredentialRetrieverIntegrationTest {
+
+  /** Tests retrieval via {@code docker-credential-gcr} CLI. */
+  @Test
+  public void testRetrieveGCR() throws IOException {
+    try {
+      DockerCredentialRetriever dockerCredentialRetriever =
+          new DockerCredentialRetriever("gcr.io", "gcr");
+
+      Authorization authorization = dockerCredentialRetriever.retrieve();
+
+      // Checks that some token was received.
+      Assert.assertTrue(0 < authorization.getToken().length());
+
+    } catch (NonexistentServerUrlDockerCredentialHelperException
+        | NonexistentDockerCredentialHelperException ex) {
+
+      Assume.assumeNoException("Skipping because docker-credential-gcr CLI not set up", ex);
```

Comment:
Again, I really think this should be a straight failure. If this happens, the build should fail and the dev should set up the credential CLI. We shouldn't be running this test without that.